net-mgmt/configsync/src/opnsense/scripts/VerbNetworks/ConfigSync/requestsaws/LICENSE.txt

```diff
@@ -0,0 +1,29 @@
+Copyright (c) 2012-2013 Paul Tax <paultax@gmail.com> All rights reserved.
```

Comment:
why not pull it in as a dependency?

Reply:
Do you mean pull in as a git submodule?

Reply:
no as a plugin dependency if possible like: https://github.com/opnsense/plugins/blob/master/www/nginx/Makefile#L4

Reply:
Cool - thanks for explaining

A few things drove me to manually include these Python modules:-
 - these modules are not available via the `pkg` system (double checked, shout out to @fichtner  because I made an embarrassing mistake about the `autossh` package last week)
 - the OPNsense system does not make available a `pip` Python package management system (leads to arbitrary code etc.)
 - as best I can tell the supplied functionality is not covered by any other existing Python module available on the system

I'm not expert with FreeBSD or OPNsense packaging, I could easily be wrong, but it seems dependencies added via the `PLUGIN_DEPENDS=` attribute of a `Makefile` need to be available via the system packaging system - which in this case does not apply.

Is there a better way that I'm missing?

Reply:
Yes, this are dependent pkg files. Are the libs also not available in rhe ports repository? If they are included, a pkg can be built.

Reply:
Hmm, okay so a bit of research turns up the following

*requestsaws*
 - this Python module is not available via the FreeBSD ports tree
 - there is an alternative module which looks to be better `py27-aws-requests-auth-0.4.0` that is available via FreeBSD ports
 - I'm happy to migrate to that library if OPNsense team are interested in adopting this library, however if we are adding packages then an altogether better option might be `py27-boto-2.48.0`

*xmltodict*
 - this Python module is available via the FreeBSD ports tree as pkg `py27-xmltodict-0.11.0`
 - the package [maintainer](https://github.com/xmj) is not the same individual as the [author](https://github.com/martinblech/xmltodict) however the change logs point to the same martinblech github source.

I don't know the OPNsense vetting process for accepting packages into the release, as I understand it to make these happen they will need to be added to the OPNsense [`ports.conf`](https://github.com/opnsense/tools/blob/master/config/18.7/ports.conf)

Reply:
this is probably the maintainer of the port (freebsd package) - in this case usually not the developer of the software. Using Ports has multiple advantages:

* better updates (libs can be updated independently)
* keeps the plugins clean

you only have to wait for it until the next opnsense update is out or build it by yourself.

Reply:
Two new issue tickets have been opened to address this request
 - https://github.com/opnsense/tools/issues/88
 - https://github.com/opnsense/tools/issues/89

Reply:
@fichtner this is for you